src/In.ProjectEKA.FHIRHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,123 @@
+namespace In.ProjectEKA.FHIRHip.DataFlow
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Net.Mime;
+    using System.Text;
+    using System.Threading.Tasks;
+    using HipLibrary.Patient;
+    using HipLibrary.Patient.Model;
+    using Hl7.Fhir.Model;
+    using Hl7.Fhir.Serialization;
+    using Hl7.Fhir.Utility;
+    using Model;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Serialization;
+    using Optional;
+    using Serilog;
+
+    public class Collect : ICollect
+    {
+        private readonly string HipSystemDataUrl;
+        private readonly HttpClient HttpClient;
+
+        public Collect(string hipSystemDataUrl, HttpClient httpClient)
+        {
+            HipSystemDataUrl = hipSystemDataUrl;
+            HttpClient = httpClient;
+        }
+
+        public async Task<Option<Entries>> CollectData(DataRequest dataRequest)
+        {
+            var bundles = new List<CareBundle>();
+            var patientData = await FindPatientsData(dataRequest);
+            var careContextReferences = patientData.Keys.ToList();
+            foreach (var careContextReference in careContextReferences)
+                foreach (var result in patientData.GetOrDefault(careContextReference))
+                {
+                    Log.Information($"Returning file: {result}");
+                    var fjp = new FhirJsonParser();
+                    bundles.Add(new CareBundle(careContextReference, fjp.Parse<Bundle>(result)));
+                }
+
+            var entries = new Entries(bundles);
+            return Option.Some(entries);
+        }
+
+        private async Task<Dictionary<string, List<string>>> FindPatientsData(DataRequest request)
+        {
+            // LogDataRequest(request);
+            
+            var patientReferenceNumber = request.CareContexts.First().PatientReference;
+            var careContexts = request.CareContexts.Select(careContext => careContext.CareContextReference).ToList();
+            var dataResponse = await GetPatientsData(new NetworkDataRequest(patientReferenceNumber,
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=6044012)